code/go/0chain.net/blobbercore/allocation/allocationchange.go

```diff
@@ -265,6 +265,11 @@ func (a *AllocationChangeCollector) MoveToFilestore(ctx context.Context) error {
 
 			for _, ref := range refs {
 
+				var count int64
+				tx.Model(&reference.Ref{}).
+					Where("allocation_id=? AND validation_root=? AND type=?", a.AllocationID, ref.PrevValidationRoot, reference.FILE).
```

Comment:
I suppose its better to add isPreCommit condition.

Reply:
I don't think so its required there, we are just checking if this prevRoot is not the current root of any other Ref, be it preCommit or not.